pkg/reconciler/ingress/ingress.go

```diff
@@ -253,6 +254,11 @@ func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ing *v1alpha1
 	// First, create all needed VirtualServices.
 	kept := sets.NewString()
 	for _, d := range desired {
+		if d.GetAnnotations()[networking.IngressClassAnnotationKey] != network.IstioIngressClassName {
```

Comment:
Do we even reach this code in a live system? Is the filtering in the informer done based on the old state of the ingress or the new one? Since we filter by istio annotation in controller.go.

Reply:
Yes, we reach this code even though the informer has the filter. The informer done based on the old state of the ingress, which still has istio annotation.

Here is the process during the annotation update.

1. User deploys Ingress w/ ` ingress.class`=`istio` annotaiton. (original virtualservice).
2. User edits the annotation in Ingress/VirtualService (desired virtualservice).
3. Reconciler creates desired VirtualService Objects.
4. Reconciler(Informer) lists existing VirtualService.
5. Reconciler replaces original VirtualService by desired VirtualService.

So, existing virtualservice still has istio annotation and Informer lists them(step-4). Then, they are updated with unexpected annotation(step-5).